api/ocp/scc.go

```diff
@@ -0,0 +1,36 @@
+package ocp
+
+import (
+	_ "embed"
+	"fmt"
+	"strings"
+
+	secv1 "github.com/openshift/api/security/v1"
+	"sigs.k8s.io/yaml"
+)
+
+//go:embed scc.yaml
+var sccYAMLTemplate string
+
+// NewSecurityContextConstraints loads the embedded SCC YAML template, replaces the namespace,
+// and returns it as a SecurityContextConstraints object
+func NewSecurityContextConstraints(name string, namespace string) (*secv1.SecurityContextConstraints, error) {
```

Comment:
what i was expecting was like below
```go
//go:embed scc.yaml
var sccYAMLTemplate string

// GetSecurityContextConstraintsTemplate returns the embedded SCC YAML template as a string.
func GetSecurityContextConstraintsTemplate() string {
	return sccYAMLTemplate
}
```

and let the user import it and unmarshal and do generate the below details

```
- system:serviceaccount:ceph-csi-operator-system:csi-rbd-ctrlplugin-sa
  - system:serviceaccount:ceph-csi-operator-system:csi-cephfs-ctrlplugin-sa
  - system:serviceaccount:ceph-csi-operator-system:csi-nfs-ctrlplugin-sa
  - system:serviceaccount:ceph-csi-operator-system:csi-rbd-nodeplugin-sa
  - system:serviceaccount:ceph-csi-operator-system:csi-cephfs-nodeplugin-sa
  - system:serviceaccount:ceph-csi-operator-system:csi-nfs-nodeplugin-sa
```

The service account and namespace and service account name is what others need to worry about.

@nb-ohad Is this what you also have in mind? @leelavg Does this works well for you?

Reply:
@Madhu-1 The question here is: why should anyone have to add a code dependency on this project to consume one static yaml template file? 

My alternative is. have the file in this project. Instruct other projects to pull this file into their project as part of their build phase using something like `curl get` directly from github (or something similar)  

Regarding the code itself. I don't believe the above is going to work. The second code base is going to pull this code as a dependency but not the temple file (As it is not a go file). This means that the embed command will fail and the project will not build. 

Unfortunately, if you want to provide it via code, we will need to embed the entire content in a multi-line const string.